File: src/main/java/com/sequenceiq/cloudbreak/converter/JsonToClusterConverter.java
package com.sequenceiq.cloudbreak.converter;

import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.controller.json.ClusterRequest;
import com.sequenceiq.cloudbreak.domain.Cluster;
import com.sequenceiq.cloudbreak.domain.Status;

@Component
public class JsonToClusterConverter extends AbstractConversionServiceAwareConverter<ClusterRequest, Cluster> {
    @Override
    public Cluster convert(ClusterRequest source) {
        Cluster cluster = new Cluster();
        cluster.setName(source.getName());
        cluster.setStatus(Status.REQUESTED);
        cluster.setDescription(source.getDescription());
        cluster.setEmailNeeded(source.getEmailNeeded());
        Boolean enableSecurity = source.getEnableSecurity();
        cluster.setSecure(enableSecurity == null ? false : enableSecurity);
        cluster.setKerberosMasterKey(source.getKerberosMasterKey());
        cluster.setKerberosAdmin(source.getKerberosAdmin());
        cluster.setKerberosPassword(source.getKerberosPassword());
        return cluster;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/JsonToClusterConverter.kt
package com.sequenceiq.cloudbreak.converter

import com.sequenceiq.cloudbreak.api.model.Status.REQUESTED

import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.api.model.AmbariStackDetailsJson
import com.sequenceiq.cloudbreak.api.model.ClusterRequest
import com.sequenceiq.cloudbreak.api.model.FileSystemBase
import com.sequenceiq.cloudbreak.api.model.RDSConfigJson
import com.sequenceiq.cloudbreak.domain.AmbariStackDetails
import com.sequenceiq.cloudbreak.domain.Cluster
import com.sequenceiq.cloudbreak.domain.FileSystem
import com.sequenceiq.cloudbreak.domain.RDSConfig

@Component
class JsonToClusterConverter : AbstractConversionServiceAwareConverter<ClusterRequest, Cluster>() {
    override fun convert(source: ClusterRequest): Cluster {
        val cluster = Cluster()
        cluster.name = source.name
        cluster.status = REQUESTED
        cluster.description = source.description
        cluster.emailNeeded = source.emailNeeded
        cluster.userName = source.userName
        cluster.password = source.password
        val enableSecurity = source.enableSecurity
        cluster.secure = enableSecurity ?: false
        cluster.kerberosMasterKey = source.kerberosMasterKey
        cluster.kerberosAdmin = source.kerberosAdmin
        cluster.kerberosPassword = source.kerberosPassword
        cluster.isLdapRequired = source.ldapRequired
        cluster.configStrategy = source.configStrategy
        val ambariStackDetails = source.ambariStackDetails
        cluster.enableShipyard = source.enableShipyard
        if (ambariStackDetails != null) {
            cluster.ambariStackDetails = conversionService.convert<AmbariStackDetails>(ambariStackDetails, AmbariStackDetails::class.java)
        }
        val rdsConfigJson = source.rdsConfigJson
        if (rdsConfigJson != null) {
            cluster.rdsConfig = conversionService.convert<RDSConfig>(rdsConfigJson, RDSConfig::class.java)
        }
        val fileSystem = source.fileSystem
        if (fileSystem != null) {
            cluster.fileSystem = conversionService.convert<FileSystem>(fileSystem, FileSystem::class.java)
        }
        return cluster
    }
}
